src/Mvc/Mvc.Razor.RuntimeCompilation/src/DependencyInjection/RazorRuntimeCompilationMvcCoreBuilderExtensions.cs

```diff
@@ -71,7 +71,7 @@ internal static void AddServices(IServiceCollection services)
             var compilerProvider = services.FirstOrDefault(f =>
                 f.ServiceType == typeof(IViewCompilerProvider) &&
                 f.ImplementationType?.Assembly == typeof(IViewCompilerProvider).Assembly &&
-                f.ImplementationType.FullName == "Microsoft.AspNetCore.Mvc.Razor.Compilation.DefaultViewCompiler");
+                f.ImplementationType.FullName == "Microsoft.AspNetCore.Mvc.Razor.Compilation.DefaultViewCompilerProvider");
```

Comment:
Offending code. I know I should do better than this, but this is what happens when I'm left to do work unchecked